client/src/NationalStats.tsx

```diff
@@ -0,0 +1,55 @@
+import { Component } from 'react';
+import CanadaNationalStats from './Models/CanadaNationalStats';
+
+export default class NationalStats extends Component<{longitude: any, latitude: any}, {canadaNationalStats: CanadaNationalStats}> {
+    async componentDidMount(): Promise<void> {
+    }
+    async componentWillReceiveProps(nextProps: {latitude: any, longitude: any}): Promise<void> {
+        if (nextProps.latitude !== 0 && nextProps.longitude !== 0) {
+            this.setState({canadaNationalStats: await this.getLocationFromPosition(nextProps.latitude, nextProps.longitude)});
+        }
+    }
+    componentWillUnmount() {
+        // fix Warning: Can't perform a React state update on an unmounted component
+        this.setState = (state,callback)=>{
+            return;
+        };
+    }
+
+    async getLocationFromPosition(latitude: number, longitude: number): Promise<any> {
+
+        try {
+            const url = `http://localhost:8080/location?longitude=${longitude.toPrecision(5)}&latitude=${latitude.toPrecision(5)}&component=nationalstats`;
+            let result: any = await (await fetch(url)).json();//.json();
+            const tStats: CanadaNationalStats = result;
+            return tStats;
+        }
+        catch (e) {
+            console.log(e);
+        }
+    }
+    
+    render() {
+        let renderNatJSON;
+        try
+        {
+        renderNatJSON = (
+        <div>
+            <h6>Canada National Stats:</h6>
+            <h6>New Cases: {this?.state?.canadaNationalStats.CasesNew}</h6>
```

Comment:
We need to find a way to handle null objects better than them just not rendering partially I think. (Needs additional discussion)

Reply:
Not totally sure what to do about this, since 90% of the data (and the components that depend on them) we return depends on (them accepting) location, which would mean that if we just fully didn't render components before we got their data, the site both before they agreed to location services or if they didn't would be almost totally empty. One thing might be to keep them rendered without data but have some dropdowns (or equivalent) to allow the user to select a location to view data for (probably a v2.0 thing, but its a thought).